test/Streamly/Test/Common/Array.hs

```diff
@@ -86,6 +85,23 @@ testLengthFromStreamN = genericTestFrom A.fromStreamN
 #ifndef TEST_SMALL_ARRAY
 testLengthFromStream :: Property
 testLengthFromStream = genericTestFrom (const A.fromStream)
+
+testFromListN :: Property
```

Comment:
Please do the following,

Check all the array modules where the API `fromListN` is present.
Wrap `fromListN` around `#if defined()` on the applicable modules.

```
TEST_SMALL_ARRAY -> Data.SmallArray
TEST_ARRAY -> Data.Array.Storable.Foreign(.Types)
DATA_ARRAY_PRIM_PINNED -> Data.Array.Prim.Pinned(.Types)
DATA_ARRAY_PRIM -> Data.Array.Prim(.Types)
*_DEFAULT* -> Data.Array
```

*Side Note*: This should be renamed better and multiple modules can be
 eliminated. I'll send a PR later for this.

For example, Say `fromListN` is defined in `Data.Array.Storable.Foreign` and
`Data.Array.Prim.Pinned`.

Then the test should be written as:

```
#if defined(TEST_ARRAY) && defined(DATA_ARRAY_PRIM_PINNED)
testFromListN = ...
#endif
```

You can read more about CPP [here](https://guide.aelve.com/haskell/cpp-vww0qd72).